scripts/zfs-tests.sh

```diff
@@ -224,6 +229,42 @@ shift $((OPTIND-1))
 FILES=${FILES:-"$FILEDIR/file-vdev0 $FILEDIR/file-vdev1 $FILEDIR/file-vdev2"}
 LOOPBACKS=${LOOPBACKS:-""}
 
+if [[ -n "$SINGLETEST" ]]; then
+	SINGLETESTDIR="$(dirname $SINGLETEST)"
+	SINGLETESTFILE="$(basename $SINGLETEST)"
+	RUNFILEDIR="/var/tmp"
+	RUNFILE="${SINGLETESTFILE}.run"
```

Comment:
It would be good to add a `.$$` to this to get a more unique name.  Even if conflicts due to concurrent users seem unlikely!  How about:

```
RUNFILE="${SINGLETESTFILE}.$$.run"
```

You'll also want to remove the file after the ZTS exits.